index.html

```diff
@@ -1,100 +1,122 @@
+            if (file) {
+                loadedFile = file;
+                alert('Plik CSV został załadowany i zapisany.');
+                populateDropdown();
+            }
+        });
+
+        document.getElementById('select').addEventListener('change', function () {
+            var selectedOption = this.value;
+            if (loadedFile) {
+                var reader = new FileReader();
+                reader.onload = function (e) {
+                    var csvData = e.target.result;
+                    var rows = csvData.split('\n');
+                    var selectedDistrict = rows[selectedOption].split(';')[0];
+
+                    var districtRows = rows.filter(function (row) {
+                    return row.startsWith(selectedDistrict + ';');
+                    });
 
+                    displayResults(districtRows);
+                    kwadratodleglosci();
+                };
 
+                reader.readAsText(loadedFile);
+            } else {
+                alert('Proszę najpierw załadować plik CSV.');
+            }
+        });
 
+        function kwadratodleglosci() {
+            var selectedOption = document.getElementById('select').value;
+            if (loadedFile && selectedOption !== "") {
+                var reader = new FileReader();
+                reader.onload = function (e) {
+                    var csvData = e.target.result;
+                    var rows = csvData.split('\n');
+                    var selectedDistrict = rows[selectedOption].split(';')[0];
+
+                    var districtRows = rows.filter(function (row) {
+                    return row.startsWith(selectedDistrict + ';');
+                    });
+
+                    var squaredDistance = 0;
+                    for (var i = 0; i < districtRows.length; i++) {
+                    var rowData = districtRows[i].split(';');
+                    var votersCount = parseFloat(rowData[4].replace(',', '.')) || 0;
+                    squaredDistance += Math.pow(votersCount, 2);
+                    }
+
+                    document.getElementById('kwadratodleglosci').innerHTML = 'Kwadrat odległości dla Okręgu Wyborczego ' + selectedDistrict + ': ' + squaredDistance;
+                };
+
+                reader.readAsText(loadedFile);
+            } else {
+                alert('Proszę najpierw załadować plik CSV i wybrać okręg.');
+            }
+        }
+
+        function populateDropdown() {
+            var reader = new FileReader();
+            reader.onload = function (e) {
+                var csvData = e.target.result;
+                var rows = csvData.split('\n');
+                var selectDropdown = document.getElementById('select');
+                selectDropdown.innerHTML = "";
+                addedDistricts = [];
+                for (var i = 0; i < rows.length; i++) {
+                    var districtNumber = rows[i].split(';')[0];
+                    if (!addedDistricts.includes(districtNumber)) {

```

Comment:
Well done figuring out a way to only have unique districts 🎉 

Minor: Instead of having it as an array, you can look into using a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set), which is a data structure that works sort of like an array, but is optimized for storing only unique items inside. This would fit perfectly for this problem. You can also ignore this comment until some later time, as it's minor and an array is totally fine for our use case 😄